src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSBuildReferenceTaxonomyUtils.java

```diff
@@ -0,0 +1,318 @@
+     * accessions from the reference that have not been successfully looked up; if null,
+     * will be initialized to the accToRefInfo key set by default.
+     * <p>
+     * Returns a collection of reference accessions that could not be found, if any.
+     */
+    protected static Set<String> parseCatalog(final BufferedReader reader,
+                                              final Map<String, Tuple2<String, Long>> accessionToNameAndLength,
+                                              final Map<String, PSPathogenReferenceTaxonProperties> taxIdToProperties,
+                                              final boolean bGenBank, Set<String> accessionsNotFound) {
+        try {
+            String line;
+            final int taxIdColumnIndex, accessionColumnIndex;
+            if (bGenBank) {
+                taxIdColumnIndex = 6;
+                accessionColumnIndex = 1;
+            } else {
+                taxIdColumnIndex = 0;
+                accessionColumnIndex = 2;
+            }
+            if (accessionsNotFound == null) {
+                accessionsNotFound = new HashSet<>(accessionToNameAndLength.keySet());
+            }
+            final int minColumns = Math.max(taxIdColumnIndex, accessionColumnIndex) + 1;
+            long lineNumber = 1;
+            while ((line = reader.readLine()) != null && !line.isEmpty()) {
+                final String[] tokens = line.trim().split("\t", minColumns + 1);
+                if (tokens.length >= minColumns) {
+                    final String taxId = tokens[taxIdColumnIndex];
+                    final String accession = tokens[accessionColumnIndex];
+                    if (accessionToNameAndLength.containsKey(accession)) {
+                        final Tuple2<String, Long> nameAndLength = accessionToNameAndLength.get(accession);
+                        updateAccessionTaxonProperties(taxId, nameAndLength._1, nameAndLength._2, taxIdToProperties);
+                        accessionsNotFound.remove(accession);
+                    }
+                } else {
+                    throw new UserException.BadInput("Expected at least " + minColumns + " tab-delimited columns in " +
+                            "GenBank catalog file, but only found " + tokens.length + " on line " + lineNumber);
+                }
+                lineNumber++;
+            }
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile("Error reading from catalog file", e);
+        }
+        return accessionsNotFound;
+    }
+
+    /**
+     * Parses scientific name of each taxon
+     */
+    protected static void parseNamesFile(final BufferedReader reader, final Map<String, PSPathogenReferenceTaxonProperties> taxIdToProperties) {
+        try {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                final String[] tokens = line.split("\t\\|\t");
```

Comment:
The regex is looking for <tab>|<tab> as the delimiter. The double backslash is required for literals, otherwise Java thinks I'm trying to use an escape character called "\|" and throws an error.